### PR TITLE
Fix for keyboard navigationwhen search-enabled is false

### PR DESCRIPTION
**Pull request by fcaballero**
There are other PR with some fix for #375 but I got some problem trying to rebase, therefor I added those changes in a clean master in order to don't have problems when merged. 
The PRs mentioned are #1114 and #1109

### CODE REVIEW

**Comment by wesleycho** [comment]
Changes to `dist` need to be undone. Please run `git reset --hard ac0c72f && git push -f`

File: src/select2/select.tpl.html [hunk 1]
```diff
@@ -6,13 +6,13 @@
...
        ng-class="{'select2-display-none': !$select.open}">
-    <div class="select2-search" ng-show="$select.searchEnabled">
+    <div class="select2-search" ng-class="{'ui-select-offscreen':!$select.searchEnabled}">
       <input type="text" autocomplete="off" autocorrect="false" autocapitalize="off" spellcheck="false"
-       role="combobox"
-       aria-expanded="true"
-       aria-owns="ui-select-choices-{{ $select.generatedId }}"
-       aria-label="{{ $select.baseTitle }}"
-       aria-activedescendant="ui-select-choices-row-{{ $select.generatedId }}-{{ $select.activeIndex }}"
+             role="combobox"
```
Comment (wesleycho) [inline src/select2/select.tpl.html]: Fix indentation to 2 spaces

Comment (fcaballero) [inline src/select2/select.tpl.html]: Done, please check if it's what you expected...

**Comment by user378230** [comment]
1. Can you rebase once more? it should get rid of a bit of noise in the test diff. 
2. Can you fix the white-spacing on the new ng-class attributes 
(add space: `ng-class="{'ui-select-search-hidden': !$select.searchEnabled}"`
3. You need to squash, when doing so please amend the message according the the commit message style guidelines in [CONTRIBUTING.md](https://github.com/angular-ui/ui-select/blob/master/CONTRIBUTING.md#commit)

**Comment by ramonjsantiago** [comment]
Can the owner of this project resolve the conflicts and merge this pull request #1543 into the angular-ui-select master please?

I really need this one specific fix.

**Comment by fcaballero** [comment]
@wesleycho, @user378230 is everything ok for merging this? or do I miss or mess up something?

**Comment by user378230** [comment]
@fcaballero I think you merged from master rather than rebased. The only commit should be the one with your changes. 

Note also 2 & 3 in my other response:
> 1. ...
> 2. Can you fix the white-spacing on the new ng-class attributes (add space: ng-class="{'ui-select-search-hidden': !$select.searchEnabled}"
> 3. You need to squash, when doing so please amend the message according the the commit message style guidelines in CONTRIBUTING.md

**Comment by ramonjsantiago** [comment]
Can yoy guys please resolve the conflicts and merge this fix in? It has been hanging out there for a while.

**Comment by user378230** [comment]
@ramonjsantiago want to have a go yourself? The first post has all the details you need :smile:

**Pull request closed without merging**